### PR TITLE
fix: added platform-specific dependencies to Pipfile

### PR DESCRIPTION
**Pull request by aleextw**
added colorama for win32 systems, updated bootstrap to remove deprecation warning

### CODE REVIEW

**Comment by issacj-17** [comment]
Updates have been tested on Windows 10 (64-bit). The web application is able to launch successfully. They have no visible points of failure when navigating apart from those to be implemented (Mini Project 2 Exercises). 

Updates have also been tested on Ubuntu 22.04.1 LTS with the following adjustments via a Virtual Machine (VMWare Workstation) -
sudo apt-get remove python3-pipenv
sudo pip3 install pipenv (or sudo pip install pipenv)

They have failed on my system. However, the cause has been identified as per these issues on github- https://github.com/pypa/pipenv/issues/5075
https://github.com/pypa/setuptools/issues/3278

In summary, it is due to the setuptools>=60.0.0 causing the virtualenv to create abnormal tree structure on ubuntu 22.04

It is therefore a OS dependent issue. This should not affect Vocareum given that it was able to run pipenv install and pipenv shell for mp_sort.

Additionally, Alex has tested Mini Project 2 on Debian and MacOS. The project is working as intended.

Summary:
Windows 10 (64-bit) : GOOD
Debian : GOOD
MacOS : GOOD

The changes are successful and should have little to no effect when running on Vocareum.

**Comment by kurniawano** [comment]
@aleextw , can you change from `Bootstrap4` to `Bootstrap5` instead? See https://bootstrap-flask.readthedocs.io/en/stable/basic/#initialization. After that I will merge the pull request.

**Comment by aleextw** [comment]
Updated Bootstrap4 to Bootstrap5 in code where necessary as per request

**Review by kurniawano — APPROVED** [review]
